src/main/java/sky/pro/java/course2/service/ExaminerServiceImpl.java

```diff
@@ -26,14 +24,21 @@ public ExaminerServiceImpl(QuestionService questionService, Random random) {
     @Override
     public Collection<Question> getQuestions(int amount) {
         int total = questionService.getAll().size();
-        if (amount <= total) {
-            amount = total;
-            for (int i = 0; i < amount; i++) {
-                return Collections.unmodifiableCollection(questionService.getRandomQuestion());
-            }
-        }
+        if (amount > total) {
             throw new OutOfQuestionsException("you require from us more questions than we have");
+        }
+        Set<Question> questionSet = new HashSet<>();
+        int i = 0;
+        while (i == amount) {
+            i++;
+            questionSet.add(questionService.getRandomQuestion());
 
         }
```

Comment:
```suggestion
        while (questionSet.size() != amount) {
            questionSet.add(questionService.getRandomQuestion());

        }
```
Уже намного лучше! Только немного поправим условие.